frappe/utils/global_search.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) 2015, Frappe Technologies Pvt. Ltd. and Contributors
+# License: GNU General Public License v3. See license.txt
+
+from __future__ import unicode_literals
+
+import frappe
+
+def setup_table():
+	'''Creates __global_seach table'''
+	if not '__global_search' in frappe.db.get_tables():
+		frappe.db.sql('''create table __global_search(
+			doctype varchar(100),
+			name varchar(140),
+			content text,
+			fulltext(content),
+			unique (doctype, name))
+			COLLATE=utf8mb4_unicode_ci
+			ENGINE=MyISAM
+			CHARACTER SET=utf8mb4''')
+	reset()
+
+def reset():
+	'''Deletes all data in __global_search'''
+	frappe.db.sql('delete from __global_search')
+
+def insert_test_contacts():
+	first_names = ["abc", "fghj", "cvbnm", "vgyui"]
+	for text in first_names:
+		frappe.get_doc(dict(doctype='Contact',first_name=text)).insert()
+	frappe.db.commit()
+
+def update_global_search(doc):
+	'''Add values marked with `in_global_search` to `frappe.flags.update_global_search` from given doc
+	:param doc: Document to be added to global search'''
+	if frappe.flags.update_global_search==None:
+		frappe.flags.update_global_search = []
+
+	content = []
+	for field in doc.meta.get_global_search_fields():
+		if getattr(field, 'in_global_search', None) and doc.get(field.fieldname):
+			if getattr(field, 'fieldtype', None) == "Table":
+				child_doctype = getattr(field, 'options', None)
+				for d in frappe.get_all(child_doctype):
```

Comment:
you don't need to do this:
`for d in doc.get(field.fieldname)` will also return you the doc